msu/hooks/entity/world/party.nut

```diff
@@ -136,148 +115,36 @@
-			}
-		}
-
-		if (this.m.Path != null)
-		{
-			if (this.m.Path.isAtWaypoint(this.getPos()))
-			{
-				this.m.Path.pop();
-
-				if (this.m.Path.isEmpty())
-				{
-					this.m.Path = null;
-					this.m.Destination = null;
-				}
-			}
-
-			if (this.m.Path != null)
-			{
-				this.m.Destination = ::World.tileToWorld(this.m.Path.getCurrent());
-			}
-		}
-
-		if (this.m.Destination != null)
+		local speedDelta = moddedSpeed * delta;
+		local move = this.move;
+		this.move = function(_dest, _speed)
 		{
-			if (this.m.IsMirrored)
-			{
-				if (this.getSprite("bodyUp").HasBrush)
-				{
-					if (this.m.Destination.Y < this.getPos().Y)
-					{
-						this.getSprite("bodyUp").Visible = false;
-						this.getSprite("body").Visible = true;
-					}
-					else
-					{
-						this.getSprite("bodyUp").setHorizontalFlipping(this.m.Destination.X < this.getPos().X);
-						this.getSprite("bodyUp").Visible = true;
-						this.getSprite("body").Visible = false;
-					}
-				}
-
-				this.getSprite("body").setHorizontalFlipping(this.m.Destination.X < this.getPos().X);
-			}
-
-			local myTile = this.getTile();
-			local speed = this.getMovementSpeed(true) * delta;
-
-			if (this.m.IsLeavingFootprints && !myTile.IsOccupied)
-			{
-				if (::Time.getVirtualTimeF() - this.m.LastFootprintTime >= 1.0)
-				{
-					local scale;
-
-					if (this.m.FootprintSizeOverride == 0.0)
-					{
-						scale = ::Math.minf(1.0, ::Math.maxf(0.4, this.m.Troops.len() * 0.05));
-					}
-					else
-					{
-						scale = this.m.FootprintSizeOverride;
-					}
-
-					::World.spawnFootprint(this.createVec(this.getPos().X - 5, this.getPos().Y - 15), this.m.Footprints[this.getDirection8To(this.m.Destination)] + "_0" + this.m.LastFootprintType, scale, this.m.FootprintSizeOverride != 0.0 ? 30.0 : 0.0, ::World.Assets.getFootprintVision(), this.m.FootprintType);
-					this.m.LastFootprintTime = ::Time.getVirtualTimeF();
-					this.m.LastFootprintType = this.m.LastFootprintType == 1 ? 2 : 1;
-				}
-			}
-
-			if (!this.move(this.m.Destination, speed))
+			if (::MSU.Mod.Debug.isEnabled("movement") && ::Math.round(speedDelta) != ::Math.round(_speed))
 			{
-				this.m.Destination = null;
+				this.testCompareMovementSpeeds(moddedSpeed, _speed)
 			}
+			return move(_dest, speedDelta);
 		}
-
-		if (this.m.IdleSoundsIndex != 0 && this.m.LastIdleSound + 10.0 < ::Time.getRealTimeF() && ::Math.rand(1, 100) <= 5 && this.isVisibleToEntity(::World.State.getPlayer(), 500))
-		{
-			this.m.LastIdleSound = ::Time.getRealTimeF();
-			::Sound.play(::Const.SoundPartyAmbience[this.m.IdleSoundsIndex][::Math.rand(0, ::Const.SoundPartyAmbience[this.m.IdleSoundsIndex].len() - 1)], ::Const.Sound.Volume.Ambience, this.getPos());
-		}
-
-		::MSU.Mod.Debug.isEnabled("movement")
-		{
-			this.testCompareMovementSpeeds()
-		}
+		onUpdate();
+		this.move = move;
 	}
 
-	o.testCompareMovementSpeeds <- function()
+	o.testCompareMovementSpeeds <- function(_moddedSpeed, _speed)
```

Comment:
Private functions should be prefixed with `__`

Reply:
sure